### ISSUE TITLE
 `fetch` and `axios.get` called from a test against an oak server, receive a 503.

### ISSUE THREAD

**Issue by jmfrank63** [oakserver/oak#678]
When calling fetch against a `deno` server setup with `oak` both `fetch` and `axios.get` receive a 503. The call made from a browser or `curl` called with `Deno.Command` correctly works.

Example response from `fetch`
```
Response {
  body: ReadableStream { locked: false },
  bodyUsed: false,
  headers: Headers {
    "cache-control": "no-cache",
    "content-length": "6909",
    "content-type": "text/html",
    date: "Thu, 17 Oct 2024 12:20:27 GMT",
    expires: "Sat, 17 Jun 2000 12:00:00 GMT",
    "last-modified": "Wed, 08 Jun 1955 12:00:00 GMT",
    pragma: "no-cache"
  },
  ok: false,
  redirected: false,
  status: 503,
  statusText: "Service Unavailable",
  url: "http://127.0.0.32:64061/health"
}
```

The scenario:

I run an `oak.listen` with added `SIGHUP` signal and signal listener. `Deno run` works and reloads correctly. 
But testing `deno test -A` runs into a 503. This is independent of the address `127.0.0.1:3090`, for example, shows the same behaviour. Also, the issue is present without sending any `SIGHUP`.  

Replacing `fetch` or `axios.get` with 

```
        const cmd = new Deno.Command("curl", {
            args: [`http://${config.host}:${config.port}/health`],
            stdout: "piped",
            stderr: "piped",
        });
        
                const { code, stdout, stderr } = await cmd.output();

        let jsonResponse;
        if (code === 0) {
            const output = new TextDecoder().decode(stdout);
            jsonResponse = JSON.parse(output);
            console.log("Response from /health endpoint:\n", jsonResponse);
        } else {
            const error = new TextDecoder().decode(stderr);
            console.error("Curl error:", error);
        }
        console.log("Expected health response ", expectedHealthResponse);
```

is my current workaround but it is rather cumbersome and depends on an external command.

```
export function startServer(config, healthResponse) {
  const { host, port, nodesFile, servicesFile } = config;

  const app = new Application();
  const router = new Router();

  // Define /health endpoint to return health information
  router.get("/health", (ctx) => {
    ctx.response.body = JSON.stringify(healthResponse, null, 2);
    ctx.response.type = "application/json";
  });

  // Add router to the app
  app.use(router.routes());
  app.use(router.allowedMethods());

  const controller = new AbortController();
  const { signal } = controller;
  // Start server
  console.log(`Starting server on http://${host}:${port}`);
  const server = app.listen({ hostname: host, port, signal });

  // Handle SIGHUP for reloading configuration
  Deno.addSignalListener("SIGHUP", async () => {
    console.log("Received SIGHUP signal, reloading configuration...");
    try {
      const nodes = await readNodes(nodesFile);
      const services = await readServices(servicesFile);

      healthResponse.nodes = nodes;
      healthResponse.services = services;
      const allNodes = [...nodes, { host, port }];
      healthResponse.leader = allNodes.reduce((prev, current) =>
        ipToNumber(prev.host) > ipToNumber(current.host) ? prev : current
      );
      healthResponse.timestamp = new Date().toISOString();
      console.log("Configuration reloaded successfully.");
    } catch (error) {
      console.error("Failed to reload configuration:", error);
    }
  });
  return { server, controller }
}
```
is the function I am testing.

I am not sure if this is expected behaviour or a bug.

```
deno --version
deno 2.0.0 (stable, release, x86_64-unknown-linux-gnu)
v8 12.9.202.13-rusty
typescript 5.6.2
```
Oak version: 
```
import { Application } from "jsr:@oak/oak/application";
import { Router } from "jsr:@oak/oak/router";```

i.e. latest. `@oak/oak@17.1.0`

No typescript is used, pure `ECMA` script.

Please let me know if you need further information.

**Comment by jmfrank63** [comment]
Well, I am not so sure any more if this is an oak problem at all. I replaced oak with `Deno.serve` and got the same behaviour. 
Even worse, I compiled my application spawned it as a subprocess, made a fetch call against it and got a 503. 
So, using a `deno` test to spawn a `deno` application and making a fetch call against its server results in a 503.
This might have to do with the fact that `deno` serve uses fetch under the hood. But from a high-level perspective, this is rather strange.
